src/core_read.cpp

```diff
@@ -18,18 +18,19 @@
 #include <boost/algorithm/string/split.hpp>
 
 #include <algorithm>
+#include <atomic>
 #include <string>
 
 namespace {
-
-opcodetype ParseOpCode(const std::string& s)
+class OpCodeParser
 {
-    static std::map<std::string, opcodetype> mapOpNames;
+private:
+    std::map<std::string, opcodetype> mapOpNames;
 
-    if (mapOpNames.empty())
+public:
+    OpCodeParser()
     {
-        for (unsigned int op = 0; op <= MAX_OPCODE; op++)
-        {
+        for (unsigned int op = 0; op <= MAX_OPCODE; op++) {
```

Comment:
nit

```suggestion
        for (unsigned int op = 0; op <= MAX_OPCODE; ++op) {
```

Reply:
Does it have any impact on codegen? I thought it's not longer the case since like 2002.

Reply:
That's why it's just a nit. "The compiler can fix it" isn't really a good reason to keep poor code.

Reply:
> Does it have any impact on codegen? I thought it's not longer the case since like 2002.

Not in the case of an integer.
Though in general it's better to write `++x` in for loops even if just to avoid these kind of code review comments :smile: